src/golang/lib/collections/shared/types.go

```diff
@@ -27,8 +27,9 @@ const (
 	CanceledExecutionStatus  ExecutionStatus = "canceled"
 	// Registered is a special state that indicates a workflow has been registered
 	// but has no workflow runs yet
-	RegisteredExecutionStatus ExecutionStatus = "registered"
-	UnknownExecutionStatus    ExecutionStatus = "unknown"
+	RegisteredExecutionStatus      ExecutionStatus = "registered"
+	UnknownExecutionStatus         ExecutionStatus = "unknown"
+	LambdaCompletedExecutionStatus ExecutionStatus = "lambda_completed"
```

Comment:
This seems a little confusing to have an engine specific status as part of our workflow and operator life cycle states. Can we move this elsewhere?